### ISSUE TITLE
php package pulls in apache dependency

### ISSUE THREAD

**Issue by wmertens** [NixOS/nixpkgs#28312]
When using the `phpfpm` service for use with nginx, it pulls in `php` which in turn pulls in `apache` and `apache-dev`. This is wholly unnecessary for the operation of phpfpm.

The culprit is this line in `/nix/store/6y6zxcj8ps4273s77c2z2g3zblrxfasm-php-7.1.5/include/php/main/build-defs.h`:

```
#define PHP_INSTALL_IT          "$(mkinstalldirs) '/nix/store/6y6zxcj8ps4273s77c2z2g3zblrxfasm-php-7.1.5/modules' &&   
              $(mkinstalldirs) '$(INSTALL_ROOT)/nix/store/k6gqdllpsvj1h7i9hj903di61ggkzc8g-apache-httpd-2.4.25/conf' &&        
          /nix/store/85x2nk3klnwb6gr9p1him18ds1pf4sqj-apache-httpd-2.4.25-dev/bin/apxs -S LIBEXECDIR='/nix/store/6y6zxcj8ps4273s77c2z2g3zblrxfasm-php-7.1.5/modules'           

-S SYSCONFDIR='$(INSTALL_ROOT)/nix/store/k6gqdllpsvj1h7i9hj903di61ggkzc8g-apache-httpd-2.4.25/conf'           

^^^^ this setting here

             -i -a -n php7 libphp7.la"
```

I don't know enough about PHP to know if this should be there or not, but in any case it is the only reference to it. It seems that elsewhere on the web that `SYSCONFDIR` is set to `/etc/php`…

## Technical details

* System: NixOS: 17.03.1463.81fceb2554 (Gorilla)

**Comment by fpletz** [comment]
This only applies to 17.03 and has been resolved on master:

```
λ yolovo nixpkgs → λ git release-17.03 → nix-store --query --requisites $(nix-build -A php) | grep apache
/nix/store/gfpfsmd9izbfg00lhm3cxx8zxh53amf3-apache-httpd-2.4.27
/nix/store/567fkr7ghb98pzwgydn9fn82vw5ll4j9-apache-httpd-2.4.27-dev

λ yolovo nixpkgs → λ git master → nix-store --query --requisites $(nix-build -A php) | grep apache
```

Haven't looked which commit is responsible for that, though.

**Comment by globin** [comment]
I'd prefer not to fix this in 17.03, it is somewhat unfortunate, but I think this can wait for 17.09 to be fixed in a stable release..
